### PR TITLE
feat(Validation): Added validation attributes

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**

### CODE REVIEW

**Review by Gusinuhe — DISMISSED** [review]

File: Runtime/Behaviors/DelayBehavior.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using System.Collections;
 using System.Runtime.Serialization;
 using Innoactive.Creator.Core.Attributes;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Behaviors/DelayBehavior.cs:5]: Is it even required?

File: Runtime/Behaviors/DisableGameObjectBehavior.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using Innoactive.Creator.Core.Attributes;
 using Innoactive.Creator.Core.SceneObjects;
 using Innoactive.Creator.Core.Utils;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Behaviors/DisableGameObjectBehavior.cs:5]: Is it even required?

File: Runtime/Behaviors/EnableGameObjectBehavior.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 using Innoactive.Creator.Core.Attributes;
 using Innoactive.Creator.Core.SceneObjects;
 using Innoactive.Creator.Core.Utils;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Behaviors/EnableGameObjectBehavior.cs:5]: Is it even required?

File: Runtime/Behaviors/HighlightObjectBehavior.cs [hunk 4]
```diff
@@ -5,6 +5,7 @@
 using Innoactive.Creator.Core.SceneObjects;
 using Innoactive.Creator.Core.Properties;
 using Innoactive.Creator.Core.Utils;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Behaviors/HighlightObjectBehavior.cs:8]: Is it even required?

File: Runtime/Behaviors/MoveObjectBehavior.cs [hunk 5]
```diff
@@ -39,6 +40,7 @@ public class EntityData : IBehaviorData
             /// <summary>
             /// Duration of the transition. If duration is equal or less than zero, target object movement is instantaneous.
             /// </summary>
+            [OptionalValue]
```
Comment (Gusinuhe) [inline Runtime/Behaviors/MoveObjectBehavior.cs:43]: Is it really optional? 🤔

File: Runtime/Behaviors/PlayAudioBehavior.cs [hunk 6]
```diff
@@ -5,6 +5,7 @@
 using Innoactive.Creator.Core.Audio;
 using Innoactive.Creator.Core.Configuration;
 using Innoactive.Creator.Core.Configuration.Modes;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Behaviors/PlayAudioBehavior.cs:8]: Is it even required?

File: Runtime/Conditions/PositionalConditions/ObjectInColliderCondition.cs [hunk 7]
```diff
@@ -3,6 +3,7 @@
 using Innoactive.Creator.Core.SceneObjects;
 using Innoactive.Creator.Core.Properties;
 using Innoactive.Creator.Core.Utils;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Conditions/PositionalConditions/ObjectInColliderCondition.cs:6]: Is it even required?

File: Runtime/Conditions/PositionalConditions/ObjectInRangeCondition.cs [hunk 8]
```diff
@@ -3,6 +3,7 @@
 using Innoactive.Creator.Core.SceneObjects;
 using Innoactive.Creator.Core.Properties;
 using Innoactive.Creator.Core.Utils;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Conditions/PositionalConditions/ObjectInRangeCondition.cs:6]: Is it even required?

File: Runtime/Conditions/TimeoutCondition.cs [hunk 9]
```diff
@@ -1,5 +1,6 @@
 using System.Runtime.Serialization;
 using Innoactive.Creator.Core.Attributes;
+using Innoactive.Creator.Core.Validation;
```
Comment (Gusinuhe) [inline Runtime/Conditions/TimeoutCondition.cs:3]: Is it even required?

**Review by Gusinuhe — COMMENTED** [review]
🤔

File: Runtime/Behaviors/MoveObjectBehavior.cs (hunk 5 above)
Comment (SimonTheSourcerer) [reply Runtime/Behaviors/MoveObjectBehavior.cs:43]: instant transition is fine for me :P

**Review by Gusinuhe — APPROVED** [review]

File: Editor/UI/Drawers/Audio Data Drawers/AudioDataDrawer.cs [hunk 10]
```diff
@@ -13,11 +12,22 @@ namespace Innoactive.CreatorEditor.Core.UI.Drawers
...
         /// <inheritdoc />
         public override Rect Draw(Rect rect, object currentValue, Action<object> changeValueCallback, GUIContent label)
         {
+            tooltip = label.tooltip;
+            image = label.image;
             // Has to run with a null label to not show two labels, dont ask me why.
             return base.Draw(rect, currentValue, changeValueCallback, GUIContent.none);
         }
+
+        protected override void CheckValidationForValue(object currentValue, MemberInfo info, GUIContent label)
```
Comment (tomwim) [inline Editor/UI/Drawers/Audio Data Drawers/AudioDataDrawer.cs:27]: inheritdoc please

File: Runtime/Behaviors/MoveObjectBehavior.cs (hunk 5 above)
Comment (tomwim) [reply Runtime/Behaviors/MoveObjectBehavior.cs:44]: Also would say 0 is fine

**Review by tomwim — CHANGES_REQUESTED** [review]

File: Runtime/Behaviors/MoveObjectBehavior.cs (hunk 5 above)
Comment (Gusinuhe) [reply Runtime/Behaviors/MoveObjectBehavior.cs:44]: Should this attribute be in a defined symbol?

**Review by Maxsiom — APPROVED** [review]

**Pull request merged**